src/Mvc/Mvc.Abstractions/src/ModelBinding/Metadata/ModelMetadataIdentity.cs

```diff
@@ -95,7 +95,9 @@ public static ModelMetadataIdentity ForParameter(ParameterInfo parameter, Type m
         ArgumentNullException.ThrowIfNull(parameter);
         ArgumentNullException.ThrowIfNull(modelType);
 
-        return new ModelMetadataIdentity(modelType, parameter.Name, fieldInfo: parameter);
+        var containerType = parameter.Member is PropertyInfo propertyInfo ? propertyInfo.DeclaringType : null;
+
+        return new ModelMetadataIdentity(modelType, parameter.Name, containerType, fieldInfo: parameter);
```

Comment:
I have to marinate on this change since this codepath is shared with MVC's model binding layer and I'm cautious about inadvertently introducing behavior changes based on this delta given that MVC doesn't have support for `[AsParameters]`. I also worry about the clarity of including logic that depends on the specifics of the `PropertyAsParameterInfo` type into this generic `ForParameters` overload.

I do see that given the set of available APIs it's difficult to directly construct the `ModelMetadataIdentity` object with the appropriate metadata. I do wonder if we should bite the bullet and include a `ModelMetadataIdentity.ForPropertyAsParameter` overload.

Reply:
I think that in this case I would opt to a solution similar to the previous one. In the case that the ParameterInfo contains a member that is a PropertyInfo then create  it with Model identity.ForProperty and in the case it is not Model identity.ForParameter.

What do you think? I do not see either adding a ForPropertyAsParametwr

Reply:
I think the ideal solution would be a mix of the two. Something like:

```csharp
private static EndpointModelMetadata CreateModelMetadata(IParameterBindingMetadata parameter, Type type)
{
    if (parameter.ParameterInfo is { } parameterInfo)
    {
		if (parameterInfo.Member is PropertyInfo propertyInfo && propertyInfo.DeclaringType is not null)
		{
        	return new(ModelMetadataIdentity.ForProperty(propertyInfo, type, propertyInfo.DeclaringType));
    	}
		return ModelMetadataIdentity.ForParameter(parameterInfo);
	}
    return CreateModelMetadata(type);
}
```

Thoughts?

Reply:
Agree! :)